src/Raven.Server/Documents/Queries/QueryBuilderHelper.cs

```diff
@@ -573,4 +576,65 @@ internal enum ComparerType
         AscendingAlphanumeric,
         DescendingAlphanumeric
     }
+    
+    internal static IShape HandleWkt(Query query, MethodExpression expression, QueryMetadata metadata, BlittableJsonReaderObject parameters, string fieldName,
+            SpatialField spatialField, out SpatialUnits units)
+        {
+            var wktValue = QueryBuilderHelper.GetValue(query, metadata, parameters, (ValueExpression)expression.Arguments[0]);
+            QueryBuilderHelper.AssertValueIsString(fieldName, wktValue.Type);
+
+            SpatialUnits? spatialUnits = null;
+            if (expression.Arguments.Count == 2)
+                spatialUnits = GetSpatialUnits(query, expression.Arguments[1] as ValueExpression, metadata, parameters, fieldName);
+
+            units = spatialUnits ?? spatialField.Units;
+
+            var wkt = CoraxGetValueAsString(wktValue.Value);
+
+            try
+            {
+                return spatialField.ReadShape(wkt, spatialUnits);
+            }
+            catch (Exception e)
+            {
+                throw new InvalidQueryException($"Value '{wkt}' is not a valid WKT value.", query.QueryText, parameters, e);
```

Comment:
can we provide any more details (what is the document id, for example)

Reply:
This is the query builder.

Reply:
I believe that once we get it as an indexing error we'll see the id of a document on which it threw. It's handled at higher level.

Reply:
I believe that once we get it as an indexing error we'll see the id of a document on which it threw. It's handled at higher level.